File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {

        int n = height.size();
        int left= 0;
        int right = n-1;
        int maxi = 0;
        while(left<right){
            if(height[left] < height[right]){
                maxi = max(maxi, height[left]*(right-left));
                left++;
            }
            else{
                maxi = max(maxi, height[right]*(right-left));
                right--;
            }

        }
        return maxi;


        
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        
        n = len(height)
        start = 0
        end = n-1
        max_output = 0
        while(start<end):
            max_output = max(max_output,
                             min(height[start], height[end]) * (end-start)
                            )
            if(height[start]<height[end]):
                start+=1
            else:
                end-=1
            
        return max_output
        